runtime/lua/vim/_meta/api_keysets_extra.lua

```diff
@@ -247,3 +247,24 @@ error('Cannot require a meta file')
 --- @field fill integer
 --- @field end_row integer
 --- @field end_vcol integer
+
+-- Inherit from vim.api.keyset.win_config so this type can be passed to nvim_open_win().
+-- Because of this we only need to define the fields with different types (nil or non-nil).
+
+--- @class vim.api.keyset.win_config_ret : vim.api.keyset.win_config
```

Comment:
I hope we don't actually need comments explaining why types inherit each other :) This is generally a good idea.

Reply:
I only added this because I specifically needed to add fields with an explicit `nil` type to essentially remove them, which I felt was a bit weird, but means this type can be passed to `nvim_open_win()`.

Usually subtypes like this just redefine the fields to remove `nil`, which I don't think needs any explanation.